### ISSUE TITLE
Equal Earth projection is missing ellipsoidal implementation

### ISSUE THREAD

**Issue by beuan** [OSGeo/PROJ#1097]
Thank you @jdugge @kbevers @rouault on your awesome work getting the Equal Earth projection so fast available in PROJ library. I am supper amazed how this is moving forward. 

The current PROJ implementation is sphere-only, but I noticed [people are starting to use it with WGS84 ellipsoid](https://github.com/geoserver/geoserver/blob/master/data/release/user_projections/epsg.properties#L32), which might cause issues later if this projection is released as this. Used with WGS84 and current implementation, it also loses its equal-area property.  

In the paper we suggest: 
> For projecting geographic coordinate on the ellipsoids, such as WGS 1984, we suggest transforming geographic latitudes to authalic latitudes (Snyder 1987, p. 16) before projecting with the spherical Equal Earth projection. (Šavrič et al. 2018)

I believe functions for this conversions already exists in PROJ code base:
```
double pj_qsfn(double sinphi, double e, double one_es)
double pj_authlat(double beta, double *APA)
```
I am not very well familiar with the whole code base, but with provided functions we might be able to add ellipsoidal implementation before its release on Sep 1st. I am willing to assist as much as possible.

I assume we need new `e_forward()` and `e_inverse()` functions, updated `PJ *PROJECTION(eqearth)` function and new tests. Would there be anything else?

Org Issue: #1085
Pull Request: #1090

cc: @berniejenny

**Comment by beuan** [comment]
I forked PROJ and implemented ellipsoidal equations for the Equal Earth projection. Unfortunately, I do not know how to run tests not using `CMAKE`. I was able to build new version and test points using `NMAKE`. Is there any other way I can verify that tests are passing? Thanks.
Forked branch with proposed fix: https://github.com/OSGeo/proj.4/compare/master...beuan:EqEarth_ellips

**Comment by beuan** [comment]
Thank you @jdugge for running the tests! Looks like your tests are not passing due to the missing tolerance (I think). 
Would you mind running them again? I just added the tolerance parameter.  I hope this fixes the issue.

**Comment by kbevers** [comment]
Closed with #1101
